x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/EvalMapper.java

```diff
@@ -73,7 +66,7 @@ private EvalMapper() {}
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
     public static Supplier<ExpressionEvaluator> toEvaluator(Expression exp, Layout layout) {
-        if (exp instanceof Mappable m) {
+        if (exp instanceof EvaluatorMapper m) {
```

Comment:
question: do we plan to align all entries from the `MAPPERS` list above, eventually, so that all expressions are simply represented as `EvaluatorMapper`s? I.e. will `EvalMapper.MAPPERS` eventually be removed?

(In the current state it's a bit confusing that e.g. `Sub` is handled as `EvaluatorMapper`, but `Neg` is not.)

Reply:
It depends on the situation - if the class is part of ESQL yes however expressions defined somewhere (which might eventually be the case) require an external mapper.